Realm/ObjectServerTests/SwiftFlexibleSyncServerTests.swift

```diff
@@ -1315,6 +1356,318 @@ extension SwiftFlexibleSyncServerTests {
+        realm.syncSession!.suspend()
+        let ex = XCTestExpectation(description: "no attempt to re-create subscription, returns immediately")
+        Task {
+            _ = try await realm.objects(SwiftPerson.self).where { $0.age >= 8 }.subscribe(name: "8 or older")
+            _ = try await results0.subscribe(name: "8 or older")
+            XCTAssertEqual(realm.subscriptions.count, 1)
+            ex.fulfill()
+        }
+        await fulfillment(of: [ex], timeout: 5.0)
+        XCTAssertEqual(realm.subscriptions.count, 1)
+    }
+
+    @MainActor
+    func testSubscribeSameQueryDifferentName() async throws {
+        try await populateSwiftPerson()
+        let realm = try openFlexibleSyncRealm()
+
+        let results0 = try await realm.objects(SwiftPerson.self).where { $0.age >= 8 }.subscribe()
+        _ = try await realm.objects(SwiftPerson.self).where { $0.age >= 8 }.subscribe(name: "8 or older")
+        _ = try await results0.subscribe(name: "older than 7")
+        XCTAssertEqual(realm.subscriptions.count, 3)
+        let first = realm.subscriptions.first(ofType: SwiftPerson.self) { $0.age >= 8 }
+        XCTAssertNil(first?.name)
+        let second = realm.subscriptions[1]
+        XCTAssertEqual(second!.name, "8 or older")
+        let third = realm.subscriptions[2]
+        XCTAssertEqual(third!.name, "older than 7")
+    }
+
+    @MainActor
+    func testSubscribeDifferentQuerySameName() async throws {
+        try await populateSwiftPerson()
+        let realm = try openFlexibleSyncRealm()
+
+        _ = try await realm.objects(SwiftPerson.self).where { $0.age > 8 }.subscribe(name: "group1")
+        _ = try await realm.objects(SwiftPerson.self).where { $0.age > 5 }.subscribe(name: "group1")
+        XCTAssertEqual(realm.subscriptions.count, 1)
+        for subscription in realm.subscriptions {
+            XCTAssertEqual(subscription.queryString, "age > 5")
+        }
+    }
+
+    @MainActor
+    func testUnsubscribe() async throws {
+        try await populateSwiftPerson()
+        let realm = try openFlexibleSyncRealm()
+
+        let results1 = try await realm.objects(SwiftPerson.self).where { $0.lastName == "lastname_3" }.subscribe()
+        XCTAssertEqual(realm.subscriptions.count, 1)
+        results1.unsubscribe()
+        XCTAssertEqual(realm.subscriptions.count, 0)
+    }
+
+    @MainActor
+    func testUnsubscribeAfterReassign() async throws {
+        try await populateSwiftPerson()
+        let realm = try openFlexibleSyncRealm()
+
+        var results0 = try await realm.objects(SwiftPerson.self).where { $0.age >= 8 }.subscribe()
+        XCTAssertEqual(results0.count, 3)
+        XCTAssertEqual(realm.subscriptions.count, 1)
+        results0 = try await results0.where { $0.age < 8 }.subscribe() // subscribes to "age >= 8 && age < 8" because that's the local query
+        XCTAssertEqual(results0.count, 0)
+        XCTAssertEqual(realm.subscriptions.count, 2) // "age >= 8" and "age >= 8 && age < 8"
```

Comment:
```suggestion
        XCTAssertEqual(realm.subscriptions.count, 2) // "age >= 8" OR "age >= 8 && age < 8"
```

Reply:
This is saying that the two subs present are 1) "age >= 8" 2) "age >= 8 && age < 8"
Not a logical AND. I can change.